test/e2e/tests/test_permission.py

```diff
@@ -0,0 +1,155 @@
+		ref,
+		period_length=DELETE_WAIT_AFTER_SECONDS,
+	)
+	assert deleted
+
+@service_marker
+@pytest.mark.canary
+class TestPermission:
+	def test_crud(self, permission):
+		res, ref = permission
+
+		# Verify state and spec have values we need to test
+		cr = k8s.get_resource(ref)
+		assert cr is not None
+		assert 'spec' in cr
+		assert 'policyTemplate' in cr['spec']
+		assert 'status' in cr
+		assert 'status' in cr['status']
+		assert cr['status']['status'] == "ATTACHABLE"
+		assert 'version' in cr['status']
+		assert cr['status']['version'] == '1'
+		assert 'ackResourceMetadata' in cr['status']
+		assert 'arn' in cr['status']['ackResourceMetadata']
+		resource_arn = cr['status']['ackResourceMetadata']['arn']
+
+		# Test updating Permission by changing policyTemplate
+		original_policy_template = {
+			"Action": ["imagebuilder:ListComponents"],
+			"Effect": "Allow"
+		}
+		permission_version = ram_permission.get_permission(resource_arn, 1)
+		policy_template = permission_version["permission"]
+		assert policy_template == json.dumps(original_policy_template)
+		assert permission_version["defaultVersion"]
+
+		updated_policy_template = {
+			'Action': ['imagebuilder:ListComponents', 'imagebuilder:GetComponent'],
+			'Effect': 'Allow'
+		}
+		updates = {
+			"spec": {
+				"policyTemplate": json.dumps(updated_policy_template)
+			}
+		}
+		k8s.patch_custom_resource(ref, updates)
+		time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+
+		# Check resource synced successfully
+		assert k8s.wait_on_condition(ref, "ACK.ResourceSynced", "True", wait_periods=5)
+
+		cr = k8s.get_resource(ref)
+		assert 'spec' in cr
+		assert 'policyTemplate' in cr['spec']
+		assert 'status' in cr
+		assert 'version' in cr['status']
+		assert cr['status']['version'] == '2'
+		assert 'defaultVersion' in cr['status']
+		assert cr['status']['defaultVersion']
+
+		permission_version = ram_permission.get_permission(resource_arn, 2)
+		policy_template = permission_version["permission"]
+		assert policy_template == json.dumps(updated_policy_template)
+		assert permission_version["defaultVersion"]
+
+		# Test reverting back to previous policyTemplate
+		updates = {
+			"spec": {
+				"policyTemplate": json.dumps(original_policy_template)
+			}
+		}
+		k8s.patch_custom_resource(ref, updates)
+		time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+
+		# Check resource synced successfully
+		assert k8s.wait_on_condition(ref, "ACK.ResourceSynced", "True", wait_periods=5)
+		
+		cr = k8s.get_resource(ref)
+		assert 'spec' in cr
+		assert 'policyTemplate' in cr['spec']
+		assert 'status' in cr
+		assert 'version' in cr['status']
+		assert cr['status']['version'] == '3'
+		assert 'defaultVersion' in cr['status']
+		assert cr['status']['defaultVersion']
+
+
+		permission_version = ram_permission.get_permission(resource_arn, 3)	
+		policy_template = permission_version["permission"]
+		assert policy_template == json.dumps(original_policy_template)
+		assert permission_version["defaultVersion"]
```

Comment:
Can we also verify that there is only one version available? a.k.a the controller deleted the old one